src/platform/FailSafeContext.cpp

```diff
@@ -85,6 +91,13 @@ void FailSafeContext::ScheduleFailSafeCleanup(FabricIndex fabricIndex, bool addN
 
 CHIP_ERROR FailSafeContext::ArmFailSafe(FabricIndex accessingFabricIndex, System::Clock::Timeout expiryLength)
 {
+    auto currentMonotonicTimeMs = System::SystemClock().GetMonotonicMilliseconds64();
+    if (!mFailSafeArmed)
+    {
+        System::Clock::Timeout maxCumulativeTimeout = System::Clock::Milliseconds64(CHIP_DEVICE_CONFIG_MAX_CUMULATIVE_FAILSAFE_SEC * 1000);
```

Comment:
what happens if this starttimer succeeds, but below startimer or commitToStorage or SetFailSafeArmed fail? Will `this` still be valid to have `FailSaveTimerExpired` on it called?